src/analysis/retail/priest/shadow/modules/talents/Deathspeaker.tsx

```diff
@@ -44,17 +47,41 @@ class Deathspeaker extends Analyzer {
       Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.DEATHSPEAKER_TALENT_BUFF),
       this.onBuffRefresh,
     );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.SHADOW_WORD_DEATH_TALENT),
+      this.onCast,
+    );
+  }
+
+  onCast() {
+    //for debuging. Sometimes chargesAvailable, and chargesOnCooldown don't correctly add up to getMaxCharges.
+    //console.log("SD CAST",this.spellUsable.chargesAvailable(TALENTS.SHADOW_WORD_DEATH_TALENT.id),"/",this.spellUsable.chargesOnCooldown(TALENTS.SHADOW_WORD_DEATH_TALENT.id),"max:",this.abilities.getMaxCharges(TALENTS.SHADOW_WORD_DEATH_TALENT.id));
   }
 
   onBuffApplied(event: ApplyBuffEvent) {
     this.procsGained += 1; // Add a proc to the counter
-    if (this.spellUsable.isOnCooldown(TALENTS.SHADOW_WORD_DEATH_TALENT.id)) {
-      this.spellUsable.endCooldown(TALENTS.SHADOW_WORD_DEATH_TALENT.id);
-    }
+    this.abilities.increaseMaxCharges(event, TALENTS.SHADOW_WORD_DEATH_TALENT.id, 1);
+    this.spellUsable.endCooldown(
+      TALENTS.SHADOW_WORD_DEATH_TALENT.id,
+      event.timestamp,
+      false,
+      false,
+    );
     this.lastProcTime = event.timestamp;
   }
 
   onBuffRemoved(event: RemoveBuffEvent) {
+    this.abilities.decreaseMaxCharges(event, TALENTS.SHADOW_WORD_DEATH_TALENT.id, 1);
+    if (this.spellUsable.chargesAvailable(TALENTS.SHADOW_WORD_DEATH_TALENT.id) === 1) {
+      // In certain circumstances, if you have 1 charge available after the buff, you can end up having negative charges spellUsable.onCooldown
+      // Resting the cooldown entirely fixes the issue.
```

Comment:
poking @kfinch about this. this is a very unique case but still maybe something he knows about

Reply:
This was happening with the Shadowy Insight before it was changed as well.  I don't know why it happens, but if you are at max charges when your charges are reduced it sometimes gives you a negative amount onCooldown.  I think having a check to make sure spellusable.OnCooldown doesn't allow for negative numbers would fix this, but I haven't looked at it.